Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -1250,29 +1250,42 @@ internal struct JSONScanner {
   /// Throws if field name cannot be parsed.
   /// If it encounters an unknown field name, it silently skips
   /// the value and looks at the following field name.
-  internal mutating func nextFieldNumber(names: _NameMap) throws -> Int? {
+  internal mutating func nextFieldNumber(
+    names: _NameMap,
+    messageType: Message.Type?,
```

Comment:
does `messageType` have to be optional, or can we require it?

Reply:
This is internal and all the callers provide it, so yes it can be non-optional.